snakemake_executor_plugin_slurm/__init__.py

```diff
@@ -83,6 +124,28 @@ class ExecutorSettings(ExecutorSettingsBase):
 # Required:
 # Implementation of your executor
 class Executor(RemoteExecutor):
+    def clean_old_logs(self) -> None:
+        """Delete files older than specified age from the SLURM log directory."""
+        # shorthands:
+        age_cutoff = self.workflow.executor_settings.delete_logfiles_older_than
+        logdir_path = Path(self.workflow.executor_settings.logdir)
+        keep_all = self.workflow.executor_settings.keep_successful_logs
+        if age_cutoff <= 0 or keep_all:
+            return
+        cutoff_secs = age_cutoff * 86400
+        current_time = time.time()
+        self.logger.info(f"Cleaning up log files older than {age_cutoff} day(s)")
+        for path in logdir_path.rglob("*"):
+            try:
+                if path.is_file():
+                    file_age = current_time - path.stat().st_mtime
+                    if file_age > cutoff_secs:
+                        path.unlink()
+                    elif path.is_dir() and not any(path.iterdir()):
+                        path.rmdir()
+            except (OSError, FileNotFoundError) as e:
+                self.logger.warning(f"Could not delete file {path}: {e}")
+
```

Comment:
Move below __post_init__ for readability?

Reply:
ah, this was a left-over from the initial tinkering, where I violated the name space, if not before post-init. Will check and change according to the suggestion, when I find the time - ought to be working now.